### PR TITLE
Fix bug in `unbind`

### LINKED ISSUE #12: `unbind` removes all handlers
I understood from the Readme that `unbind`, given a handler that was previously bound to the server, should only remove the respective handler. However, `unbind` currently unbinds all handlers that have been bound previously:

```js
mailServer.bind(handler1);
mailServer.bind(handler2);

mailServer.unbind(handler2);
```

This will result in `handler1` not being called. If wanted, I'll do a PR that fixes this behaviour 🛠

### PR DESCRIPTION
**Pull request by fallafeljan**
Fixes #12.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -121,12 +121,13 @@ module.exports = {
 				});
 			},
 			unbind: function(rcpt,handler) {
-				var tmp = [];
+				var tmp = {};
```
Comment (fallafeljan) [inline lib/index.js]: This was the main issue, which caused `handlers` to be an empty array after calling `unbind` once

File: test/test.js [hunk 2]
```diff
@@ -125,13 +129,13 @@ test('two emails with different handlers', function(t) {
 			sendmail(recp, subject+"0", body+"0", cb);
 		},
 		function (res,cb) {
-			mailServer.bind(recp,handler0(cb));
+			mailServer.bind(recp,handler0Factory(cb));
```
Comment (fallafeljan) [inline test/test.js]: Calling `handler*factory` now assigns the created handler directly to `handler*`, which passes the equality check in `unbind`. Before, different function instances caused the equality check (`f !== g`) to fail.

**Comment by deitch** [comment]
Oh Lordy, seriously? What a silly mistake! Nice catch, though, thank you. Checking now.

**Comment by deitch** [comment]
In and published. Thanks!

**Pull request merged**